modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -199,6 +200,66 @@ public class ReplicaManager extends AbstractEventProducer<LocalReplicaEvent, Loc
 
     private volatile String localNodeConsistentId;
 
+    /* Temporary converter to support the zone based partitions in tests. **/
+    private Function<ReplicationGroupId, ReplicationGroupId> groupIdConverter = Function.identity();
+
+    /**
+     * Constructor for a replica service.
+     *
+     * @param nodeName Node name.
+     * @param clusterNetSvc Cluster network service.
+     * @param cmgMgr Cluster group manager.
+     * @param clockService Clock service.
+     * @param messageGroupsToHandle Message handlers.
+     * @param placementDriver A placement driver.
+     * @param requestsExecutor Executor that will be used to execute requests by replicas.
+     * @param idleSafeTimePropagationPeriodMsSupplier Used to get idle safe time propagation period in ms.
+     * @param failureProcessor Failure processor.
+     * @param raftCommandsMarshaller Command marshaller for raft groups creation.
+     * @param raftGroupServiceFactory A factory for raft-clients creation.
+     * @param raftManager The manager made up of songs and words to spite all my troubles is not so bad at all.
+     * @param volatileLogStorageFactoryCreator Creator for {@link org.apache.ignite.internal.raft.storage.LogStorageFactory} for
+     *      volatile tables.
+     * @param groupIdConverter Temporary converter to support the zone based partitions in tests.
+     */
+    @TestOnly
+    public ReplicaManager(
```

Comment:
Is it expected to be removed after full feature merge?

Reply:
Yep, made a TODO